File: LinkedList/LinkedList.py
class Node:
    def __init__(self, initData):
        self.data = initData
        self.next = None

    def setData(self, newData):
        self.data = newData

    def setNext(self, newNext):
        self.next = newNext


class LinkedList:
    def __init__(self):
        self.head = None

    def getHead(self):
        return self.head.data

    def isEmpty(self):
        return self.head == None

    def add(self, data):
        temp = Node(data)
        temp.setNext(self.head)
        self.head = temp

    def search(self, data):
        if self.head.data == data:
            return True
        pointer = self.head
        while(pointer):
            if pointer.data == data:
                return True
            pointer = pointer.next

    def size(self):
        count = 0
        pointer = self.head
        while(pointer):
            count += 1
            pointer = pointer.next
        return count
    def remove(self, data):
        pointer = self.head
        previousNode = None
        while(pointer):
            if pointer.data == data:
                previousNode.next = pointer.next
            previousNode = pointer
            pointer = pointer.next



ll = LinkedList()
ll.add(54)
ll.add(26)
ll.add(93)
ll.add(17)
ll.add(77)
ll.add(31)

print ll.getHead()
print ll.search(26)
ll.remove(93)
print ll.size()

File: LinkedList/LinkedList.js

class Node{
  constructor(element){
    this.element = element;
    this.next = null;
  }
};

class LinkedList{
  constructor(){
    this.length = 0;
    this.head = null;
  }
  size() {
    return this.length;
  }
  getHead() {
    return this.head;
  }
  isEmpty(){
    return this.length === 0;
  }
  indexOf(element){
    let index = 0;
    let current = this.head;
    while(current){
      index++;
      if( current.element === element )
        return index-1;
      current = current.next;
    }
    return -1;
  }
  add(element){
    const node = new Node(element);
    if (this.head === null) {
      this.head = node;
    } else {
      let current = this.head;
      while (true) {
        if (current.next === null) {
          current.next = node;
          this.length++;
          return;
        }
        current = current.next;
      }
    }
  }
  remove(element) {
    if (this.head === null) {
      return null;
    }
    let currentNode = this.head;
    let previousNode = null;
    if (currentNode.element === element) {
      head = currentNode.next;
    } else {
      while (currentNode.element !== element) {
        previousNode = currentNode;
        currentNode = currentNode.next;
      }
      previousNode.next = currentNode.next;
    }
  }
  iterateRecursively(k) {
    return this.iterateRecursivelyUtil(this.head);
  }
  iterateRecursivelyUtil(node) {
    if (node.next === null) {
      return node;
    }
    return this.iterateRecursivelyUtil(node.next);
  }
}


const ll = new LinkedList();
ll.add('mahak');
ll.add('jeff');
ll.add('matt');
ll.add('jason');
ll.add('fin');
ll.remove('matt');
// console.log(ll.getHead());
// console.log(ll.indexOf('matt'));
console.log(ll.iterateRecursively());